frontend/apps/reader/modules/readermenusearch.lua

```diff
@@ -0,0 +1,118 @@
+local CheckButton = require("ui/widget/checkbutton")
+local Event = require("ui/event")
+local InfoMessage = require("ui/widget/infomessage")
+local InputDialog = require("ui/widget/inputdialog")
+local KeyValuePage = require("ui/widget/keyvaluepage")
+local TouchMenu = require("ui/widget/touchmenu")
+local UIManager = require("ui/uimanager")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local Utf8Proc = require("ffi/utf8proc")
+local ffiUtil  = require("ffi/util")
+local _ = require("gettext")
+local T = ffiUtil.template
+
+local ReaderMenuSearch = WidgetContainer:extend{
+    kv = nil, -- key-value pairs for search results
+    search_for = nil,
+}
+
+function ReaderMenuSearch:init()
+    if self.ui then
+        self.ui.menu:registerToMainMenu(self)
+    end
+    self.search_for = G_reader_settings:readSetting("menu_search_string", _("Help"))
+    self.animation_time_s = G_reader_settings:readSetting("menu_search_animation_time_s", 1.0)
+end
+
+function ReaderMenuSearch:addToMainMenu(menu_items)
+    menu_items.search_menu = {
+        text = _("Menu Search"),
+        callback = function()
+            local search_dialog
+            search_dialog = InputDialog:new{
+                title = _("Search menu entry"),
+                description = _([[Search for a menu entry containing the following text (case insensitive).
+
+Attention: Lua patterns are used. If you want to search for '%' or '.' you have to enter '%%' or '%.']]),
```

Comment:
Not sure it's worth adding such technical info, to a menu item that is aimed at newcomers.